src/main/java/com/sngular/kloadgen/extractor/extractors/ProtoBufExtractor.java

```diff
@@ -20,25 +20,50 @@
 import com.squareup.wire.schema.internal.parser.OneOfElement;
 import com.squareup.wire.schema.internal.parser.ProtoFileElement;
 import com.squareup.wire.schema.internal.parser.TypeElement;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.RandomUtils;
 import org.apache.commons.lang3.StringUtils;
 
-public class ProtoBufExtractor {
+public class ProtoBufExtractor implements Extractor {
 
   public static final String ARRAY_POSTFIX = "-array";
 
   public static final String MAP_POSTFIX = "-map";
 
   public static final String UNSUPPORTED_TYPE_OF_VALUE = "Something Odd Just Happened: Unsupported type of value";
 
-  public final List<FieldValueMapping> processSchema(final ProtoFileElement schema) {
+  public final List<FieldValueMapping> processSchema(final Object schema) {
     final List<FieldValueMapping> attributeList = new ArrayList<>();
     final HashMap<String, TypeElement> nestedTypes = new HashMap<>();
-    schema.getTypes().forEach(field -> processField(field, attributeList, schema.getImports(), true, nestedTypes));
+    final ProtoFileElement protoFileElement = ((ProtobufSchema) schema).rawSchema();
+    protoFileElement.getTypes().forEach(field -> processField(field, attributeList, protoFileElement.getImports(), true, nestedTypes));
     return attributeList;
   }
 
+  @Override
+  public final List<FieldValueMapping> processApicurioParsedSchema(final Object schema) {
+    final List<FieldValueMapping> attributeList = new ArrayList<>();
+    final var protoFileElement = ((io.apicurio.registry.utils.protobuf.schema.ProtobufSchema) schema).getProtoFileElement();
+    // todo: en vez de null comprobar si hay que crear un HashMap con field.getNestedTypes()
+    protoFileElement.getTypes().forEach(field -> processField(field, attributeList, protoFileElement.getImports(), false, null));
+    return attributeList;
+  }
+
+  @Override
+  public final List<FieldValueMapping> processConfluentParsedSchema(final Object schema) {
+    final List<FieldValueMapping> attributeList = new ArrayList<>();
+    final var protoFileElement = ((ProtobufSchema) schema).rawSchema();
+    // todo: en vez de null comprobar si hay que crear un HashMap con field.getNestedTypes()
+    protoFileElement.getTypes().forEach(field -> processField(field, attributeList, protoFileElement.getImports(), false, null));
+    return attributeList;
+  }
+
```

Comment:
Please check the configuration of your IDE, for some reason it doesn't put the `Override` annotation on the last method in the extractor implementation

Reply:
It seems to work fine, if I remove the "final" in the definition, it send and advise.
Even testing other interface implementations (like AvroSRLoadGenerator), there is no override tag and I see no advices.
Are you sure that rules is included?